### PR TITLE
Osd 22457 add manual cordon node check

### PR DESCRIPTION
**Pull request by tkong-redhat**
Add pre-flight health check for manually cordoned worker node

### CODE REVIEW

File: pkg/upgraders/healthcheck_nodes.go [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	cops := &client.ListOptions{
+		Raw: &metav1.ListOptions{
+			LabelSelector: "node-role.kubernetes.io/worker, !node-role.kubernetes.io/infra",
+		},
+	}
+	// Get the list of worker nodes
+	err := c.List(context.TODO(), nodes, cops)
+	if err != nil {
+		logger.Info("Unable to fetch node list")
+		metricsClient.UpdateMetricHealthcheckFailed(ug.Name, metrics.ClusternNodesManuallyCordoned)
```
Comment (ravitri) [inline pkg/upgraders/healthcheck_nodes.go:29]: This metric is a bit misleading. If for some reason the API itself is not available, we need to set a different reason for failure like "healthcheck_node_query_failed" like in https://github.com/openshift/managed-upgrade-operator/blob/master/pkg/upgraders/healthcheck_alerts.go#L33 for metric query failing. Here it's possible that all the kubelets in the nodes are fine but due to API issue the list of nodes doesn't work.

Mainly, the "reason" label for healthcheck metric should correctly reflect where the failure happened.

Comment (tkong-redhat) [reply pkg/upgraders/healthcheck_nodes.go:29]: Thank you, Ravi. Make sense, I'll make the update.

**Comment by ravitri** [comment]
/label tide/merge-method-squash
/lgtm
/approve

Thanks a lot Tony! :+1:

**Pull request merged**